Change to enable running two unit tests on MacOS

seatim opened the pull request:
This PR makes a small change to a sleep time in one function that is used by two unit tests.  That change enables one unit test to pass that was otherwise failing on MacOS.  Changing the sleep time does not compromise the integrity of the test, I believe.  With all tests now passing on MacOS, the code to skip two tests is not needed, so that code is removed.

Commit:
Increase sleep time in run_interactive_interrupt and run tests on MacOS

In my testing on MacOS, the "test_interactive_interrupt" test passed without
any changes needed, and the "test_headless_interrupt" test only needed to have
this sleep time increased to pass.

boothby commented on tests/test_lib.py:
```diff
@@ -904,7 +872,7 @@ def run_interactive_interrupt(interactive):
         target=_long_running_successful_problem, args=(interactive,))
     p.start()
-    time.sleep(.1)
+    time.sleep(1)
```
See line 858, you might bump the `_long_running_successful_problem` to time out after 2s

seatim replied:
I made that change locally and tested just now on both Mac and Linux, and the tests still pass.

seatim replied:
Added that to the PR.

boothby replied:
Ah, this is a ridiculously complex test.  Line 893:
```python
    if time.perf_counter() - t0 > .5:
        # be a little generous here... but the caller should kill this in way less than .5s
        sys.exit(1)
```
I guess we should bump that to 1.5?

seatim replied:
> Ah, this is a ridiculously complex test. 

😆 It sure is!

> I guess we should bump that to 1.5?

Tested that locally and it ~~is safe~~ seems safe and prudent.  Will do.

Commit:
Use different sleep time in test depending on platform

The previous commit caused a unit test to fail on Linux.  This change fixes
that.

Commit:
Increase timeout value passed to find_embedding() call in test

Commit:
Raise time threshold for exit condition in unit test

boothby commented:
This ticket closes #175.  Thanks for your help, @seatim.

Commit:
Skip two unit tests on win32 platform
